src/common/base/ConcurrentLRUCache.h

```diff
@@ -0,0 +1,130 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#ifndef COMMON_BASE_CONCURRENTLRUCACHE_H_
+#define COMMON_BASE_CONCURRENTLRUCACHE_H_
+
+#include "base/Base.h"
+#include "base/StatusOr.h"
+#include <boost/compute/detail/lru_cache.hpp>
+#include <gtest/gtest_prod.h>
+
+namespace nebula {
+
+template<typename K, typename V>
+class ConcurrentLRUCache final {
+    FRIEND_TEST(ConcurrentLRUCacheTest, SimpleTest);
+
+public:
+    explicit ConcurrentLRUCache(size_t capacity, uint32_t bucketsExp = 4)
+        : bucketsNum_(1 << bucketsExp)
+        , bucketsExp_(bucketsExp) {
+        CHECK(capacity > bucketsNum_ && bucketsNum_ > 0);
+        auto capPerBucket = capacity >> bucketsExp;
+        auto left = capacity;
+        for (uint32_t i = 0; i < bucketsNum_ - 1; i++) {
+            buckets_.emplace_back(capPerBucket);
+            left -= capPerBucket;
+        }
+        CHECK_GT(left, 0);
+        buckets_.emplace_back(left);
+    }
+
+    bool contains(const K& key, int32_t hint = -1) {
+        return buckets_[bucketIndex(key, hint)].contains(key);
+    }
+
+    void insert(const K& key, const V& val, int32_t hint = -1) {
+        buckets_[bucketIndex(key, hint)].insert(key, val);
+    }
+
+    StatusOr<V> get(const K& key, int32_t hint = -1) {
+        return buckets_[bucketIndex(key, hint)].get(key);
+    }
+
+    /**
+     * Insert the {key, val} if key not existed, and return Status::Inserted.
+     * Otherwise, just return the value for the existed key.
+     * */
+    StatusOr<V> putIfAbsent(const K& key, const V& val, int32_t hint = -1) {
+        return buckets_[bucketIndex(key, hint)].putIfAbsent(key, val);
+    }
+
+    void clear() {
+        for (auto i = 0; i < bucketsNum_; i++) {
+            buckets_[i].clear();
+        }
```

Comment:
is it need buckets_.clear() at here?

Reply:
When we call clear,  what we want is to clear the content in cache, not destroy the buckets.